### PR TITLE
Scrapper, Sofya Kostina - 21FPL2

### PR DESCRIPTION
**Pull request by d07l**

### CODE REVIEW

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
It is right time to start implementing lab_5!

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -1,82 +1,194 @@
...
+        return self._headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
     Delivers a response from a request
     with given configuration
     """
-    pass
+    #time.sleep(random.randint(2,5))
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: is this needed or no?

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -90,25 +202,37 @@ def __init__(self, config: Config) -> None:
...
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        all_links_bs = article_bs.find_all('a')
+        for link_bs in all_links_bs:
+            link = link_bs.get('href')
+            if link is None:
+                continue
+            elif link[16:26] == '/news/2023':
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: can you simplify this check?

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
You have implementation for score 4 according to your target score. It means that if you pass lab handover, you still get no more than 3. May be you want to increase target score and write some additional logic?

File: lab_5_scrapper/scrapper.py (hunk 1 above)
Comment (d07l) [reply lab_5_scrapper/scrapper.py]: no, removed

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (d07l) [reply lab_5_scrapper/scrapper.py]: yes, I did it

File: lab_5_scrapper/scrapper.py (hunk 1 above)
Comment (demid5111) [reply lab_5_scrapper/scrapper.py]: keep

**Comment by vitazaytseva** [comment]
Well done! You can proceed to work on Lab 6)

**Pull request closed without merging**